wrappers/react/src/helpers.tsx

```diff
@@ -48,66 +45,54 @@ export function warn(...args) {
  *
  * @param {React.ReactNode} children HotTable children array.
  * @param {String} type Either `'hot-renderer'` or `'hot-editor'`.
- * @returns {Object|null} A child (React node) or `null`, if no child of that type was found.
+ * @returns {boolean} `true` if the child of that type was found, `false` otherwise.
  */
-function getChildElementByType(children: React.ReactNode, type: 'hot-renderer' | 'hot-editor'): React.ReactElement | null {
+function hasChildElementOfType(children: React.ReactNode, type: 'hot-renderer' | 'hot-editor'): boolean {
   const childrenArray: React.ReactNode[] = React.Children.toArray(children);
-  const childrenCount: number = React.Children.count(children);
-  let wantedChild: React.ReactNode | null = null;
-
-  if (childrenCount !== 0) {
-    if (childrenCount === 1 && (childrenArray[0] as React.ReactElement).props[type]) {
-      wantedChild = childrenArray[0];
 
-    } else {
-      wantedChild = childrenArray.find((child) => {
+  return childrenArray.some((child) => {
       return (child as React.ReactElement).props[type] !== void 0;
   });
-    }
-  }
-
-  return (wantedChild as React.ReactElement) || null;
-}
-
-/**
- * Get the reference to the original editor class.
- *
- * @param {React.ReactElement} editorElement React element of the editor class.
- * @returns {Function} Original class of the editor component.
- */
-export function getOriginalEditorClass(editorElement: HotEditorElement) {
-  if (!editorElement) {
-    return null;
-  }
-
-  return editorElement.type.WrappedComponent ? editorElement.type.WrappedComponent : editorElement.type;
 }
 
 /**
  * Create an editor portal.
  *
  * @param {Document} doc Document to be used.
- * @param {React.ReactElement} editorElement Editor's element.
+ * @param {React.ForwardRefExoticComponent | React.ForwardRefRenderFunction} editorComponent Editor component or render function.
+ * @param {React.RefObject} hooksReference Reference to be filled by the Editor component pointing to overridden hooks object.
  * @returns {React.ReactPortal} The portal for the editor.
  */
-export function createEditorPortal(doc: Document, editorElement: HotEditorElement): React.ReactPortal | null {
-  if (typeof doc === 'undefined' || editorElement === null) {
+export function createEditorPortal(doc: Document, editorComponent: HotTableProps['editor'] | undefined, hooksReference: React.RefObject<HotEditorHooks>): React.ReactPortal | null {
+  if (typeof doc === 'undefined' || !editorComponent) {
     return null;
   }
 
+  let Editor: React.ForwardRefExoticComponent<React.RefAttributes<HotEditorHooks>>;
+
+  if (typeof editorComponent === 'function') {
+    Editor = React.forwardRef(editorComponent as React.ForwardRefRenderFunction<HotEditorHooks>);
+  } else {
+    Editor = editorComponent as React.ForwardRefExoticComponent<React.RefAttributes<HotEditorHooks>>;
+  }
+
+  const editorElement = <Editor ref={hooksReference} />;
```

Comment:
with the suggestion in my other comment, we can then avoid passing the ref to the Editor altogether as it will come via context
```suggestion
  const editorElement = <Editor  />;
```